src/utils/c_string_util.c

```diff
@@ -19,6 +25,22 @@ char *strncpy_or_abort(char *dest, const char *src, size_t n) {
     return dest;
 }
 
+char *omf_basename(char *s) {
+    char *p = strrchr(s, PATH_DELIMITER);
+    if(p != NULL) {
+        memmove(s, p + 1, strlen(p + 1) + 1);
+    }
+    return s;
+}
+
+char *omf_dirname(char *s) {
+    char *p = strrchr(s, PATH_DELIMITER);
+    if(p != NULL) {
+        *p = '\0';
+    }
+    return s;
+}
```

Comment:
Looking at the man page, it has this example:
```
              path       dirname   basename
              /usr/lib   /usr      lib
              /usr/      /         usr
              usr        .         usr
              /          /         /
              .          .         .
              ..         .         ..
```
So my implementation is clearly not there yet.

Reply:
Well, now these tests pass, but the code looks horrible and I will see if there is something more that could be done here.